lib/modules/datasource/rubygems/get.ts

```diff
@@ -182,4 +207,113 @@ export class InternalRubyGemsDatasource extends Datasource {
 
     return new Marshal(response.body).parsed as T;
   }
+
+  private nexusEndPointRepoFrom(url: string): {
+    nexusRegistry?: string;
+    repository?: string;
+  } {
+    const endPointRe = regEx(
+      `^(?<nexusRegistry>.*:\\d{1,5}).*\\/repository\\/(?<repository>\\w+)\\/?`
+    );
+    const { nexusRegistry, repository } = url.match(endPointRe)?.groups ?? {};
+    return { nexusRegistry, repository };
+  }
+
+  @cache({
+    namespace: 'datasource-nexus',
+    key: (nexusEndPoint: string) => `isNexusDataSource:${nexusEndPoint}`,
+    ttlMinutes: 1440,
+  })
+  async isNexusDataSource(nexusEndPoint: string): Promise<boolean> {
+    const statusEndPoint = '/service/rest/v1/status';
+    const nexusUrl = nexusEndPoint.concat(statusEndPoint);
+    let isNexus = false;
+    try {
+      const response = await this.http.getJson<HttpResponse>(nexusUrl);
+      isNexus = response.headers?.server?.includes('Nexus') ?? false;
+    } catch (err) {
+      return false;
+    }
+    return isNexus;
```

Comment:
```suggestion
    try {
      const response = await this.http.getJson<HttpResponse>(nexusUrl);
      return response.headers?.server?.startsWith('Nexus/') ?? false;
    } catch (err) {
      if (err.response?.headers?.server?.startsWith('Nexus/')) {
        this.handleGenericErrors(err);
      }
      return false;
    }
```

if we get an error and it's nexus, we should rethrow, otherwise errors like 500 or 401 get silently droped